lib/msf/core/exploit/remote/ldap.rb

```diff
@@ -52,15 +57,66 @@ def get_connect_opts
         }
       end
 
-      if datastore['BIND_DN']
+      case datastore['Auth']
+      when KERBEROS_AUTH
+        kerberos_authenticator = Msf::Exploit::Remote::SMB::Client::MicrosoftKerberosAuthenticator.new(
+          host: datastore['DomainControllerIP'],
+          hostname: datastore['RHOSTNAME'],
+          domain: datastore['DOMAIN'],
+          username: datastore['USERNAME'],
+          password: datastore['PASSWORD'],
+          timeout: 25, # datastore['timeout']
+          framework: framework,
+          framework_module: self
+        )
+
+        kerberos_result = kerberos_authenticator.authenticate
+
+        connect_opts[:auth] = {
+          method: :sasl,
+          mechanism: 'GSS-SPNEGO',
+          initial_credential: kerberos_result[:security_blob],
+          challenge_response: true
+        }
```

Comment:
Do we get to verify anywhere that the spnego result is as expected? For smb I've got an extra call for that here to validate the gss response:

https://github.com/rapid7/metasploit-framework/blob/ab55abd7714091d4d1712f9c33cc22d405e2b70c/lib/msf/core/exploit/remote/smb/client/kerberos_authentication.rb#L46

But for mssql, there's a generic success/fail message that's returned without any gss specific structure - so it didn't call that functionality. I'm not sure what sasl does, if it needs a separate kerberos-specific pattern or not